api/types/database.go

```diff
@@ -965,6 +977,29 @@ func (d *DatabaseV3) IsEqual(i Database) bool {
 	return false
 }
 
+// handleAlloyDBConfig validates AlloyDB configuration
+func (d *DatabaseV3) handleAlloyDBConfig() error {
```

Comment:
I would also validate the port in the URI.
It must be either 5433 or absent from the URI (which I'm not sure we allow)